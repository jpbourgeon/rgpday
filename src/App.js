import React from 'react'
import { Router, Location } from '@reach/router'
import { withStyles } from '@material-ui/core/styles'
import { loadReCaptcha } from 'recaptcha-v3-react'
import CssBaseline from '@material-ui/core/CssBaseline'
import withRoot from 'src/withRoot'
import Loadable from 'src/components/Loadable'
import Navigation from 'src/components/Navigation'
import ProtectedRoutes from 'src/ProtectedRoutes'

const ReCaptcha = (props) => (<Loadable loadablePath='components/ReCaptcha/GetReCaptchaTokens' {...props} />)
const Default = (props) => (<Loadable loadablePath='pages/Default' {...props} />)
const Home = (props) => (<Loadable loadablePath='pages/Home' {...props} />)
const About = (props) => (<Loadable loadablePath='pages/About' {...props} />)
const Contact = (props) => (<Loadable loadablePath='pages/Contact' {...props} />)

const styles = () => ({
  '@global': {
    '.grecaptcha-badge': {
      visibility: 'hidden'
    }
  }
})

const minified = [/\/dashboard\/presentation/, /\/dashboard\/serious-game/]
const paper = [/\/dashboard\/serious-game/]
const board = [/(\/dashboard\/serious-game\/board)(.*)/]
const faded = [/\/dashboard\/presentation/]

class App extends React.Component {
  componentDidMount () {
    loadReCaptcha({
      key: '6LedLpMUAAAAAG8Ai4M4x9wTcIs4rPmvYV82a7Yh',
      id: 'rgpday.com'
    })
  }

  render () {
    return (
      <React.Fragment>

        <CssBaseline />

        <Location>
          {({ location }) => {
            return (
              <Navigation
                minified={minified.some((item) => (location.pathname.match(item)))}
                board={board.some((item) => (location.pathname.match(item)))}
                paper={paper.some((item) => (location.pathname.match(item)))}
                faded={faded.some((item) => (location.pathname.match(item)))}
                location={location}
              />
            )
          }}
        </Location>

        <Router>
          <Default default />
          <Home path='/' />
          <About path='about' />
          <Contact path='/contact' />
          <ProtectedRoutes path='/dashboard/*' />
          <ReCaptcha path='/recaptcha' />
        </Router>

      </React.Fragment>
    )
  }
}

export default withRoot(withStyles(styles)(App))
