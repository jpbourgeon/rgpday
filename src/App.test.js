/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import App from './App'
import jssSerializer from '../utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The App component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
