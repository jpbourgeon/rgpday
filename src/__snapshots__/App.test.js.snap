// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App component should render correctly 1`] = `
<div
  className="root-0"
>
  <header
    className="root-20 elevation4-26 root-11 positionFixed-12 colorPrimary-18 mui-fixed appBar-3"
  >
    <div
      className="root-47 regular-49 toolbar-1"
    >
      <button
        aria-label="Open drawer"
        className="root-130 root-133 colorInherit-134 menuButton-4"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="label-138"
        >
          <svg
            aria-hidden="true"
            className="root-139"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="root-188"
        />
      </button>
      <h1
        className="root-51 h6-68 colorInherit-80 noWrap-77 title-5"
      >
        RGPDay
      </h1>
    </div>
  </header>
  <nav
    className="drawer-6"
  >
    <div
      className="root-148 docked-149"
    >
      <div
        className="root-20 elevation0-22 paper-150 drawerPaper-7 paperAnchorLeft-151 paperAnchorDockedLeft-155"
      >
        <div
          className="toolbarIcon-2"
        >
          <button
            className="root-130 root-133"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="label-138"
            >
              <svg
                aria-hidden="true"
                className="root-139"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              className="root-188"
            />
          </button>
        </div>
        <hr
          className="root-160"
        />
        <ul
          className="root-165 padding-166"
        >
          <div>
            <div
              className="root-130 root-169 default-172 gutters-177 button-178"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <div
                className="root-181"
              >
                <svg
                  aria-hidden="true"
                  className="root-139"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 14h4v-4H4v4zm0 5h4v-4H4v4zM4 9h4V5H4v4zm5 5h12v-4H9v4zm0 5h12v-4H9v4zM9 5v4h12V5H9z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </div>
              <div
                className="root-182"
              >
                <span
                  className="root-51 body1-60 primary-185"
                >
                  Menu A
                </span>
              </div>
              <span
                className="root-188"
              />
            </div>
            <div
              className="root-130 root-169 default-172 gutters-177 button-178"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <div
                className="root-181"
              >
                <svg
                  aria-hidden="true"
                  className="root-139"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 6.92l-1.41-1.41-2.85 3.21C15.68 6.4 12.83 5 9.61 5 6.72 5 4.07 6.16 2 8l1.42 1.42C5.12 7.93 7.27 7 9.61 7c2.74 0 5.09 1.26 6.77 3.24l-2.88 3.24-4-4L2 16.99l1.5 1.5 6-6.01 4 4 4.05-4.55c.75 1.35 1.25 2.9 1.44 4.55H21c-.22-2.3-.95-4.39-2.04-6.14L22 6.92z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </div>
              <div
                className="root-182"
              >
                <span
                  className="root-51 body1-60 primary-185"
                >
                  Menu B
                </span>
              </div>
              <span
                className="root-188"
              />
            </div>
          </div>
        </ul>
        <hr
          className="root-160"
        />
        <ul
          className="root-165 padding-166"
        >
          <div>
            <div
              className="root-130 root-169 default-172 gutters-177 button-178"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <div
                className="root-181"
              >
                <svg
                  aria-hidden="true"
                  className="root-139"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </div>
              <div
                className="root-182"
              >
                <span
                  className="root-51 body1-60 primary-185"
                >
                  Menu C
                </span>
              </div>
              <span
                className="root-188"
              />
            </div>
          </div>
        </ul>
      </div>
    </div>
  </nav>
  <main
    className="content-10"
  >
    <div
      className="appBarSpacer-9"
    />
    <div
      className="root-87"
    >
      <h4
        className="root-51 h4-66 gutterBottom-78"
      >
        Material-UI
      </h4>
      <h6
        className="root-51 subtitle1-69 gutterBottom-78"
      >
        example project
      </h6>
      <button
        className="root-130 root-104 contained-115 containedSecondary-117 raised-118 raisedSecondary-120"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="label-105"
        >
          Super Secret Password
        </span>
        <span
          className="root-188"
        />
      </button>
    </div>
  </main>
</div>
`;
